control_plane/src/storage.rs

```diff
@@ -310,72 +323,112 @@ impl PageServerNode {
             .send()?
             .error_from_body()?;
         Ok(())
     }
 
     pub fn tenant_list(&self) -> Result<Vec<TenantInfo>> {
         Ok(self
-            .http_request(Method::GET, format!("{}/{}", self.http_base_url, "tenant"))
+            .http_request(Method::GET, format!("{}/tenant", self.http_base_url))
             .send()?
             .error_from_body()?
             .json()?)
     }
 
-    pub fn tenant_create(&self, tenantid: ZTenantId) -> Result<()> {
-        Ok(self
-            .http_request(Method::POST, format!("{}/{}", self.http_base_url, "tenant"))
+    pub fn tenant_create(
+        &self,
+        new_tenant_id: Option<ZTenantId>,
+    ) -> anyhow::Result<Option<ZTenantId>> {
+        let tenant_id_string = self
+            .http_request(Method::POST, format!("{}/tenant", self.http_base_url))
             .json(&TenantCreateRequest {
-                tenant_id: tenantid,
+                new_tenant_id: new_tenant_id.map(HexZTenantId::from),
             })
             .send()?
             .error_from_body()?
-            .json()?)
+            .json::<Option<String>>()?;
+
+        tenant_id_string
+            .map(|id| {
+                id.parse().with_context(|| {
+                    format!(
+                        "Failed to parse tennat creation response as tenant id: {}",
+                        id
+                    )
+                })
+            })
+            .transpose()
     }
 
-    pub fn branch_list(&self, tenantid: &ZTenantId) -> Result<Vec<BranchInfo>> {
-        Ok(self
+    pub fn timeline_list(&self, tenant_id: &ZTenantId) -> anyhow::Result<Vec<TimelineInfo>> {
+        let timeline_infos: Vec<TimelineInfoResponse> = self
             .http_request(
                 Method::GET,
-                format!("{}/branch/{}", self.http_base_url, tenantid),
+                format!("{}/tenant/{}/timeline", self.http_base_url, tenant_id),
             )
             .send()?
             .error_from_body()?
-            .json()?)
-    }
+            .json()?;
 
-    pub fn branch_create(
-        &self,
-        branch_name: &str,
-        startpoint: &str,
-        tenantid: &ZTenantId,
-    ) -> Result<BranchInfo> {
-        Ok(self
-            .http_request(Method::POST, format!("{}/branch", self.http_base_url))
-            .json(&BranchCreateRequest {
-                tenant_id: tenantid.to_owned(),
-                name: branch_name.to_owned(),
-                start_point: startpoint.to_owned(),
-            })
-            .send()?
-            .error_from_body()?
-            .json()?)
+        timeline_infos
+            .into_iter()
+            .map(TimelineInfo::try_from)
+            .collect()
     }
 
-    pub fn branch_get_by_name(
+    pub fn timeline_create(
         &self,
-        tenantid: &ZTenantId,
-        branch_name: &str,
-    ) -> Result<BranchInfo> {
-        Ok(self
+        tenant_id: ZTenantId,
+        new_timeline_id: Option<ZTimelineId>,
```

Comment:
I think, automatic generation is beneficial in cases when the client does not know what that id is or is unable to generate it. 
Also, this is an attempt to make it less heavy to do "create some timeline" call now, when tenant creation does not create create it along.
Not sure how important that is in our context, but console does generate ids and uses them, so that part is fine.

I have no strong preference, so can change it to require the ids if you insist.

Reply:
I'm not insisting, but it may be a good idea to add note about that to the api doc